kerastuner/applications/efficientnet.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Hypertunable version of ResNet based on Keras.applications."""
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+from tensorflow.keras import layers
+from tensorflow.keras.applications import efficientnet
+from tensorflow.keras.layers.experimental import preprocessing
+
+from kerastuner.engine import hypermodel
+
+
+EFFICIENTNET_MODELS = {
+    'B0': efficientnet.EfficientNetB0,
+    'B1': efficientnet.EfficientNetB1,
+    'B2': efficientnet.EfficientNetB2,
+    'B3': efficientnet.EfficientNetB3,
+    'B4': efficientnet.EfficientNetB4,
+    'B5': efficientnet.EfficientNetB5,
+    'B6': efficientnet.EfficientNetB6,
+    'B7': efficientnet.EfficientNetB7,
+}
+
+
+class HyperEfficientNet(hypermodel.HyperModel):
+    """An EfficientNet HyperModel.
+
+      # Arguments:
+
+        input_shape: shape tuple, e.g. `(256, 256, 3)`.
+              Input images will be resized if different from
+              the default input size of the version of 
+              efficientnet base model used.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        input_tensor: Keras tensor to use as image input for the model.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        classes: number of classes to classify images into.
+        augmentation: optional Model or HyperModel for image augmentation.
```

Comment:
Let's rename this to augmentation_model, since it is expected to be a model instance.

Reply:
Thanks for the suggestion.